src/combinatorics/simple_graph/regularity/chunk.lean

```diff
@@ -0,0 +1,550 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.regularity.bound
+import combinatorics.simple_graph.regularity.equitabilise
+import combinatorics.simple_graph.regularity.uniform
+
+/-!
+# Chunk of the increment partition for Szemerédi Regularity Lemma
+
+In the proof of Szemerédi Regularity Lemma, we need to partition each part of a starting partition
+to increase the energy. This file defines those partitions of parts and shows that they locally
+increase the energy.
+
+This entire file is internal to the proof of Szemerédi Regularity Lemma.
+
+## Main declarations
+
+* `szemeredi_regularity.chunk`: The partition of a part of the starting partition.
+* `szemeredi_regularity.edge_density_chunk_uniform`: `chunk` does not locally decrease the edge
+  density between uniform parts too much.
+* `szemeredi_regularity.edge_density_chunk_not_uniform`: `chunk` locally increases the edge density
+  between non-uniform parts.
+
+## TODO
+
+Once ported to mathlib4, this file will be a great golfing ground for Heather's new tactic
+`rel_congr`.
```

Comment:
Should this have any references? Even if its the same reference in every file, referring to which part of the reference the file is using seems useful.

Reply:
The entire proof (as in the entire `combinatorics.simple_graph.regularity` folder) comes from the same two pages proof, so I don't really see how to point out which part of the proof it's from.

Reply:
I think referencing the same two pages in each file is still convenient for a reader who stumbles across the file in isolation and isn't aware of those two pages.

Reply:
What do you do about the fact that our reference is a screenshot of a photo of a handout Bhavik got given in a lecture 4 years ago?

Reply:
Did bhavik publish a typeset version of his notes on github?if so, eferencing those and letting those reference the handout seems better than nothing.

Reply:
Let's reference [our ITP paper](https://drops.dagstuhl.de/opus/volltexte/2022/16718/).